tailwind/ui/buttons/Button.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from "react"
+import { cva, type VariantProps } from "class-variance-authority"
+import { Slot } from "@radix-ui/react-slot"
+
+import { BaseLink, type LinkProps } from "@/components/Link"
+
+import { cn } from "@/lib/utils/cn"
+import { type MatomoEventOptions, trackCustomEvent } from "@/lib/utils/matomo"
+import { scrollIntoView } from "@/lib/utils/scrollIntoView"
+
+const buttonVariants = cva(
+  "inline-flex gap-2 items-center justify-center rounded border-[1px] border-solid border-current text-primary transition focus-visible:outline focus-visible:outline-4 focus-visible:outline-primary-hover focus-visible:-outline-offset-1 disabled:text-disabled disabled:pointer-events-none hover:text-primary-hover [&[data-secondary='true']]:text-body",
+  {
+    variants: {
+      variant: {
+        solid:
+          "!text-background bg-primary border-transparent disabled:bg-disabled disabled:text-background hover:text-background hover:bg-primary-hover hover:shadow-button-hover active:shadow-none",
+        outline: "hover:shadow-button-hover active:shadow-none",
+        ghost: "border-transparent",
+        link: "border-transparent font-bold underline py-0 px-1 active:text-primary",
+      },
+      size: {
+        md: "min-h-10.5 px-4 py-2 [&>svg]:text-2xl",
+        sm: "text-xs min-h-[31px] py-1.5 px-2 [&>svg]:text-md",
```

Comment:
Noted that sometimes the icon shrinks and doesn't respect the desired font-size.

![image](https://github.com/user-attachments/assets/3bba3210-b729-4288-aa17-fea0381f738c)

Perhaps we could add `[&>svg]:flex-shrink-0` to the base styles to avoid this.

Reply:
Makes sense here!